### PR TITLE
Changes to enable CONSISTENCY_CHECKING

### PR DESCRIPTION
**Pull request by BrentHoltsclaw**
Signed-off-by: brentholtsclaw <brent.holtsclaw@intel.com>

### CODE REVIEW

**Review by frinzell — APPROVED** [review]

File: chipsec/hal/iobar.py [hunk 1]
```diff
@@ -68,12 +68,23 @@ def get_IO_BAR_base_address( self, bar_name ):
         if 'register' in bar:
             bar_reg   = bar['register']
             if 'base_field' in bar:
-                base_field = bar['base_field']
-                base = self.cs.read_register_field( bar_reg, base_field, preserve_field_position=True )
-                empty_base = self.cs.get_register_field_mask( bar_reg, base_field, preserve_field_position=True )
+                try:
+                    base = self.cs.read_register_field( bar_reg, base_field, preserve_field_position=True )
```
Comment (npmitche) [inline chipsec/hal/iobar.py:72]: I don't see where base_field is defined. Did you mean bar['base_field']?

File: chipsec/hal/iobar.py [hunk 2]
```diff
@@ -68,12 +68,23 @@ def get_IO_BAR_base_address( self, bar_name ):
...
+                except Exception:
+                    base = 0
+                try:
+                    empty_base = self.cs.get_register_field_mask( bar_reg, base_field, preserve_field_position=True )
```
Comment (npmitche) [inline chipsec/hal/iobar.py:76]: Same as above.

File: chipsec/hal/mmio.py [hunk 3]
```diff
@@ -179,9 +179,14 @@ def get_MMIO_BAR_base_address(self, bar_name):
             bar_reg = bar['register']
             reg_def = self.cs.get_register_def(bar_reg)
             if 'base_field' in bar:
-                base_field = bar['base_field']
-                base = self.cs.read_register_field(bar_reg, base_field, preserve_field_position=True)
-                reg_mask = self.cs.get_register_field_mask(bar_reg, base_field, preserve_field_position=True)
+                try:
+                    base = self.cs.read_register_field(bar_reg, base_field, preserve)
```
Comment (npmitche) [inline chipsec/hal/mmio.py:183]: Same

**Review by npmitche — CHANGES_REQUESTED** [review]
Doesn't look like the variable base_field is defined.

File: chipsec/hal/iobar.py (hunk 1 above)
Comment (BrentHoltsclaw) [reply chipsec/hal/iobar.py:73]: Good catch, fixed

**Review by npmitche — APPROVED** [review]

**Review by Sae86 — APPROVED** [review]

**Pull request merged**